src/internal/common/idname/idname.go

```diff
@@ -41,68 +40,72 @@ type Cacher interface {
 	ProviderForName(id string) Provider
 }
 
-var _ Cacher = &Cache{}
+var _ Cacher = &cache{}
 
-// Cache holds a cache of id-name mappings.
-type Cache struct {
+type cache struct {
 	idToName map[string]string
 	nameToID map[string]string
 }
 
-func NewCache(idToName map[string]string) Cache {
+func NewCache(idToName map[string]string) cache {
+	if len(idToName) == 0 {
+		// in case of nil
+		idToName = map[string]string{}
+	}
```

Comment:
do we need to ensure the map is non-nil? The golang [playground](https://play.golang.com/p/h00HGR73sz0) indicates that both `[]` operator and the maps package can handle a nil map

Reply:
yeah, I guess not.  Thanks, golang.